store/rootmulti/store.go

```diff
@@ -1006,19 +1011,10 @@ func (rs *Store) loadCommitStoreFromParams(key types.StoreKey, id types.CommitID
 		panic("recursive MultiStores not yet supported")
 
 	case types.StoreTypeIAVL:
-		var store types.CommitKVStore
-		var err error
-
-		if params.initialVersion == 0 {
-			store, err = iavl.LoadStore(db, rs.logger, key, id, rs.iavlCacheSize, rs.iavlDisableFastNode, rs.metrics)
-		} else {
-			store, err = iavl.LoadStoreWithInitialVersion(db, rs.logger, key, id, params.initialVersion, rs.iavlCacheSize, rs.iavlDisableFastNode, rs.metrics)
-		}
-
+		store, err := iavl.LoadStoreWithOpts(db, rs.logger, key, id, params.initialVersion, rs.iavlCacheSize, rs.iavlDisableFastNode, rs.metrics, iavltree.AsyncPruningOption(!rs.iavlSyncPruning))
```

Comment:
If this is the main place we use `!rs.iavlSyncPruning` should we just have an option that is `rs.iavlAsyncPruning` instead?

Reply:
I'm not sure what you mean

Reply:
lets discuss sync

Reply:
ok dug more here, this has to do with default values. we shouldn't really ever expect anyone to set this value, and i've updated comments to reflect that [here](https://github.com/cosmos/cosmos-sdk/pull/24090/commits/a16c63f985f547c0206c6b002d6943ce3525cd72).

that means the default value `false` will always take effect, enabling the async behavior to be default. 

in short, we want the naming of the flag to reflect what the default value implies.